### PR TITLE
Add operators table.aggr and table.join

### PR DESCRIPTION
**Pull request by mudathirmahgoub**

### CODE REVIEW

File: src/api/cpp/cvc5_kind.h [hunk 1]
```diff
@@ -3738,6 +3738,8 @@ enum Kind : int32_t
    *   - Solver::mkOp(Kind, const std::vector<uint32_t>&) const
    */
   TABLE_PROJECT,
+  TABLE_AGGREGATE,
```
Comment (ajreynol) [inline src/api/cpp/cvc5_kind.h:3741]: Need documentation here.

**Review by ajreynol — CHANGES_REQUESTED** [review]
The reduction for TABLE_AGGREGATE should be made available in the rewriter, otherwise I don't think we are evaluating TABLE_AGGREGATE yet, right?

**Comment by mudathirmahgoub** [comment]
> The reduction for TABLE_AGGREGATE should be made available in the rewriter, otherwise I don't think we are evaluating TABLE_AGGREGATE yet, right?

In prewriting, Table_AGGREGATE is reduced using bag.partition operator (which would be better to reduce it using table.group) during preprocessing 
https://github.com/cvc5/cvc5/pull/8681/files#diff-baf1cbe358ed5c34628c6afdfc771eb4fab401c8bd98b4504f6ac3c3ed651ce4R109
. So the rewriter would not see it, but it would be evaluated

File: src/api/cpp/cvc5_kind.h (hunk 1 above)
Comment (mudathirmahgoub) [reply src/api/cpp/cvc5_kind.h:3741]: Done.

File: src/theory/bags/bag_reduction.h [hunk 2]
```diff
@@ -95,9 +95,9 @@ class BagReduction : EnvObj
    *   cardinality: Int -> Int is an uninterpreted function
    *   unionDisjoint: Int -> (Bag T1) is an uninterpreted function
    */
-  Node reduceCardOperator(Node node, std::vector<Node>& asserts);
+  static Node reduceCardOperator(Node node, std::vector<Node>& asserts);
 
- private:
+  static Node reduceAggregateOperator(Node node);
```
Comment (ajreynol) [inline src/theory/bags/bag_reduction.h:100]: Add a quick doc here.

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor request else LGTM

File: src/theory/bags/bag_reduction.h (hunk 2 above)
Comment (mudathirmahgoub) [reply src/theory/bags/bag_reduction.h:112]: Done

**Review by ajreynol — APPROVED** [review]

**Pull request merged**